app/src/main/java/org/simple/clinic/scheduleappointment/facilityselection/PatientFacilityChangeEvents.kt

```diff
@@ -0,0 +1,26 @@
+package org.simple.clinic.scheduleappointment.patientFacilityTransfer
+
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.location.LocationUpdate
+import org.simple.clinic.util.RuntimePermissionResult
+import org.simple.clinic.widgets.UiEvent
+
+data class PatientFacilityChangeClicked(val facility: Facility) : UiEvent {
+  override val analyticsName = "Facility Change:Facility Clicked"
+}
+
+data class PatientFacilityLocationPermissionChanged(val result: RuntimePermissionResult) : UiEvent {
+  override val analyticsName = "Facility Change:Location Permission Changed to $result"
+}
+
+data class PatientFacilitySearchQueryChanged(val query: String) : UiEvent {
+  override val analyticsName = "Facility Change:Facility Search Query Changed"
+}
+
+data class PatientFacilityUserLocationUpdated(val location: LocationUpdate) : UiEvent {
+  override val analyticsName = "Facility Change:Location Updated ${when (location) {
```

Comment:
This can be made easier to read by extracting the when block into a separate property. But if you would like to keep it like this, that is fine with me too.

Reply:
I think this is fine for now. We are not going to add anything else to this `data class` anyway.